### ISSUE TITLE
Vertical wrapping

### ISSUE THREAD

**Issue by jacksonrayhamilton** [hernejj/workspace-grid-cinnamon-applet#14]
When I enable the "Allow cycling through workspaces" setting for Workspaces, when using this plugin, I can cycle past the left and right boundaries but not the top and bottom. I'd like to be able to cycle vertically as well.

**Comment by dadreggors** [comment]
This would make the applet feel more consistent for sure. It is a pain to have wrap implemented and working only in one direction.

**Comment by hernejj** [comment]
Yep, agreed. The problem is that the Gnome libraries, and Cinnamon, have abandoned the grid based workspaces approach. Much of the code is there there for vertical switching and my extension is able to make use of it. But every release things keep breaking :(

**Comment by dadreggors** [comment]
Since your code is based off Frippery I believe you might be able to get some good examples from Gnome's workspace-grid. I believe it is also based loosely on Frippery originally. Also, it still implements this in current Gnome and has working code. Agreed on the breakages Gnome keeps introducing with new versions but maybe they will settle soon on big code changes.:-(

https://github.com/zakkak/workspace-grid

**Comment by hernejj** [comment]
@ddreggors, Thanks for the idea. At some point I should re-sync up with Frippery and see what I've missed over the years :)
